### PR TITLE
win,poll: Add MSAFD GUID for Windows unix sockets

### PR DESCRIPTION
**Pull request by roflcopter4**
I'm typically hesitant to open a pull request for a relatively major project like this, given that I'm some idiot with close to no credentials and probably less skill, but this might just be trivial enough to pass the bar.

It annoyed me that there is no support for the new-ish (3 years now?) Windows Unix sockets in libuv. I have to admit that I don't use Windows much - I kind of hate it and everything about it's API - but that's more or less exactly why I hoped there would be support for Unix sockets. It's nice to be able to simply use the same code, no changes, etc. Why else would Microsoft add them, after all? (IIRC, they were added to stable Windows in 2020, so it's also possible that the devs were at home with nothing better to do; see also their new pre-processor of that year).

Actually adding them properly would take some major changes that would probably break everything, not that I haven't tried (see paragraph 1). But there's one simple thing that I think wouldn't break anything. As mentioned in the title, that's to add the GUID for Unix sockets to the short list in `win/poll.c` so the "fast poll" mechanism will work. It therefore must be opened manually beforehand and configured as a `uv_poll_t`. Previously this did work but defaulted to using `select`.

I don't have any tests for this at the moment. I'm not sure whether there's even the slightest interest in doing this, so I'll just make this basic request for now. I can, however, offer the supremely useless vote of confidence in saying "it worked for me".

Note: I got the GUID value by debugging the libuv code that compares them.
Note 2: I wish the official style used uppercase hexadecimal (with lowercase 0x). Oh well.

### CODE REVIEW

File: src/win/poll.c [hunk 1]
```diff
@@ -34,7 +34,9 @@ static const GUID uv_msafd_provider_ids[UV_MSAFD_PROVIDER_COUNT] = {
   {0xf9eab0c0, 0x26d4, 0x11d0,
       {0xbb, 0xbf, 0x00, 0xaa, 0x00, 0x6c, 0x34, 0xe4}},
   {0x9fc48064, 0x7298, 0x43e4,
-      {0xb7, 0xbd, 0x18, 0x1f, 0x20, 0x89, 0x79, 0x2a}}
+      {0xb7, 0xbd, 0x18, 0x1f, 0x20, 0x89, 0x79, 0x2a}},
+  {0xa00943d9, 0x9c2e, 0x4633,
+      {0x9b, 0x59, 0x00, 0x57, 0xa3, 0x16, 0x09, 0x94}}
 };
```
Comment (vtjnash) [inline src/win/poll.c:40]: Could we add comments to these to inform future travelers to these parts. From the descriptions, I see:
https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-wsaenumprotocolsw

E70F1AA0-AB8B-11CF-8CA3-00805F48A192
MSAFD Tcpip [TCP/IP][UDP/IP][RAW/IP]

F9EAB0C0-26D4-11D0-BBBF-00AA006C34E4
MSAFD Tcpip [TCP/IPv6][UDP/IPv6][RAW/IPv6]

9FC48064-7298-43E4-B7BD-181F2089792A
MSAFD L2CAP [Bluetooth]
MSAFD RfComm [Bluetooth]

A00943D9-9C2E-4633-9B59-0057A3160994
AF_UNIX

**Comment by vtjnash** [comment]
Thanks @roflcopter4!

**Comment by lazka** [comment]
Just fyi, this somehow broke ABI on 32bit Windows, making cmake and automoc crash. Rebuilding against a newer libuv fixes things though, so likely this can be ignored at this point.

See https://github.com/msys2/MINGW-packages/pull/13364#issuecomment-1264585152

**Comment by bnoordhuis** [comment]
Oh, hum. That's regrettable. I'm really not sure how this PR could cause ABI breakage but I see you've bisected it to this PR. I assume reverting would break ABI again?

**Comment by vtjnash** [comment]
I see it, and can look closer tomorrow

**Comment by vtjnash** [comment]
Apparently what we do here is a bunch of complicated nonsense, where it makes several poll handles, when only one is needed. This seems to have been fixed in the wepoll code (which was originally derived from the libuv code), but not here: https://github.com/piscisaureus/wepoll/commit/dc895c6decbadf6b9626b829304365ba802f3a35

**Comment by roflcopter4** [comment]
I thought the code looked pretty inscrutable, but I figured that either I must be too stupid to understand (usually a good assumption) or it's just Windows being Windows (also a good assumption).

**Pull request merged**